### PR TITLE
Fix nonroot user tierstore no permission bug

### LINKED ISSUE #174: [BUG] alluxio runtime to run with non-root user  has no permission to cache data
**What is your environment(Kubernetes version, Fluid version, etc.)**

**Describe the bug**

1. Create alluxio runtime

```
spec:
  replicas: 2
#  alluxioVersion:
#    image: registry.cn-huhehaote.aliyuncs.com/alluxio/alluxio
#    imageTag: "2.3.0-SNAPSHOT-f83f51e"
#    imagePullPolicy: Always
  initUsers:
    image: registry.cn-hangzhou.aliyuncs.com/fluid/init-users
    imageTag: v0.3.0
    imagePullPolicy: Always
  tieredstore:
    levels:
      - mediumtype: SSD
        path: /var/lib/docker/alluxio
        quota: 2Gi
        high: "0.95"
        low: "0.7"
  runAs:
    uid: 1005
    gid: 1005
    user: myuser
    group: mygroup
```

2. I found the alluxio worker doesn't have permission to create cache directory

```
2020-09-20 09:47:26,568 INFO  BlockWorkerFactory - Creating alluxio.worker.block.BlockWorker
2020-09-20 09:47:26,644 ERROR StorageTier - Unable to initialize storage directory at /var/lib/docker/alluxio/alluxioworker: Failed to create folder /var/lib/docker/alluxio/alluxioworker
```

3. For now, I have to add permission before creating alluxio runtime in specified node

```
chmod -R 777 /var/lib/docker/alluxio
```

**What you expect to happen:**

I hope we can give the permission 777 to the cache directory ` /var/lib/docker/alluxio`  in the init containers.

**How to reproduce it**

**Additional Information**

### PR DESCRIPTION
**Pull request by TrafalgarZZZ**
### Ⅰ. Describe what this PR does
When user attempts to run Alluxio in a non-root mode, the `tieredstore` path may be unaccessible for the non-root user due to access permission issue. For example, a directory with mode 0755,  owner `root` and group `root`. Nonroot user can't create a subdirectory under it.

**This PR run `chmod -R 0777 <tierstore-path>` in advance in init container to avoid such problem.**

### Ⅱ. Does this pull request fix one issue?

Fixes #174 

### Ⅳ. Describe how to verify it
Follow [hostpath demo](https://github.com/fluid-cloudnative/fluid/blob/master/docs/zh/samples/hostpath.md). Set tieredstore path with a limited permission directory for your non-root user.

### Ⅴ. Special notes for reviews

### CODE REVIEW

File: pkg/ddc/alluxio/types.go [hunk 1]
```diff
@@ -51,6 +51,8 @@ type Alluxio struct {
 	UFSVolumes []UFSVolume `yaml:"ufsVolumes,omitempty"`
 
 	InitUsers InitUsers `yaml:"initUsers,omitempty"`
+
+	InitTierpaths InitTierpaths `yaml:"initTierPaths,omitempty"`
```
Comment (cheyang) [inline pkg/ddc/alluxio/types.go:55]: 是否可以做为InitUsers的一个attribute？ tieredPath ？

File: pkg/ddc/alluxio/transform.go [hunk 2]
```diff
@@ -161,6 +161,13 @@ func (e *AlluxioEngine) transformCommonPart(runtime *datav1alpha1.AlluxioRuntime
 
 	value.Tieredstore.Levels = levels
 
+	// Init Tier store paths when running as a non-root user
+	// chmod -R 0777 <path>
+	paths := []string{}
+	for _, level := range levels {
+		paths = append(paths, level.Path)
+	}
+	value.InitTierpaths.Paths = strings.Join(paths, ":")
```
Comment (cheyang) [inline pkg/ddc/alluxio/transform.go:170]: 如果是MEM，是否还需要加入到这个Path中？

**Review by cheyang — CHANGES_REQUESTED** [review]

File: pkg/ddc/alluxio/types.go (hunk 1 above)
Comment (TrafalgarZZZ) [reply pkg/ddc/alluxio/types.go]: 已修改

File: pkg/ddc/alluxio/transform.go (hunk 2 above)
Comment (TrafalgarZZZ) [reply pkg/ddc/alluxio/transform.go]: 保留现有实现

**Pull request merged**